test/Microsoft.Extensions.DiagnosticAdapter.Test/DiagnosticSourceAdapterTest.cs

```diff
@@ -79,6 +90,19 @@ public void IsEnabled_FalseForNonenlistedEvent()
         }
 
         [Fact]
+        public void CallingWriteWithWrongTypeThrows()
```

Comment:
Really the detail here is that it's not just the 'wrong type' it's that it's a value type and we try to create a proxy for it.

I suspect we could benefit from beefing up our tests around value types.